tools/atomic_wget.sh

```diff
@@ -0,0 +1,23 @@
+#!/bin/bash
+
+# SPDX-License-Identifier: MPL-2.0
+
+TARGET_FILE=$(basename "$1")
+TEMP_FILE="/tmp/$TARGET_FILE"
+
+# Check if $3 is provided and not empty
+if [ -n "$3" ]; then
+    WGET_OPTIONS="$3"
+else
+    WGET_OPTIONS=""
+fi
+
+wget --continue $WGET_OPTIONS -O "$TEMP_FILE" "$2" || { \
+    echo "Error: Failed to download $1."; \
+    exit 1; \
+}
+
+mv "$TEMP_FILE" "$1" || { \
+    echo "Error: Failed to move $TEMP_FILE to $1."; \
+    exit 1; \
+}
```

Comment:
This script is not written in a production-grade quality. I ask ChatGPT to revise it and then modify a few locations manually. See the revised version.

```bash
#!/bin/bash

# SPDX-License-Identifier: MPL-2.0

# Ensure the script is invoked with the necessary arguments
if [ "$#" -lt 2 ]; then
    echo "Atomic File Download with wget."
    echo "This script downloads a file from a specified URL using wget. \
        The downloaded file will appear locally as a complete file or not appear at all in case of an error."
    echo ""
    echo "Usage: $0 <output_file> <url> [wget_options]"
    echo ""
    echo "Arguments:"
    echo "  <output_file>   Name of the file to save the downloaded content."
    echo "  <url>           URL of the file to be downloaded."
    echo "  [wget_options]  Optional: Additional wget options for customized download behavior."
    exit 1
fi

OUTPUT_FILE="$1"
URL="$2"
WGET_OPTIONS="${3:-}"

TEMP_FILE=/tmp/$(mktemp)

# Download the file using wget
if ! wget $WGET_OPTIONS -O "$TEMP_FILE" "$URL"; then
    echo "Error: Failed to download $URL."
    rm -f "$TEMP_FILE"
    exit 1
fi

# Move the temporary file to the target location
if ! mv "$TEMP_FILE" "$OUTPUT_FILE"; then
    echo "Error: Failed to move $TEMP_FILE to $OUTPUT_FILE."
    rm -f "$TEMP_FILE"
    exit 1
fi

echo "File downloaded successfully to $OUTPUT_FILE."
```

Improvements made:
* Input Validation: Added a check to ensure the script is invoked with the necessary arguments.
* Temporary File Handling: Used mktemp to create a secure temporary file.
* Conditional Check and Cleanup: Enhanced error handling to remove the temporary file if downloading or moving the file fails.
* Readability: Improved variable names and added explanatory comments for better readability.
* Optional Arguments Handling: Simplified handling of optional wget options using parameter expansion.

Reply:
Thank you for your patient instructions! I will check the script in detail before the next commit.